### PR TITLE
Feature/transition frontier diff

### PR DESCRIPTION
**Pull request by wu-s-john**
Created tests that make sure that the writes are successful to the database. Need to refine the tests such that they are able to query from these databases.

Checklist:

- [ ] Tests were added for the new behavior
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them:

### CODE REVIEW

File: src/app/archive/lib/types.ml [hunk 1]
```diff
@@ -252,3 +137,245 @@ module User_command = struct
...
+        some @@ Ledger_hash.to_string @@ Staged_ledger_hash.ledger_hash
+        @@ Blockchain_state.staged_ledger_hash blockchain_state
+
+      method global_slot = some global_slot
+
+      (* TODO: Need to implement *)
+      method ledger_proof_nonce = some 0
+
+      (* TODO: Need to implement *)
+      method status = some 0
```
Comment (cmr) [inline src/app/archive/lib/types.ml]: how needed are these?

File: scripts/archive/ocaml/stitch_introspection.ml [hunk 2]
```diff
@@ -72,35 +67,32 @@ let remove_blocks_fee_transfers_input =
...
 let read_from_standard_input () =
   (* There is only one async call in this entire program so it's okay to make this synchronous call *)
-  let json_string = Core.In_channel.input_all Core.In_channel.stdin in  
-  let json = Yojson.Basic.from_string json_string
-    |> change_constraint 
-    |> remove_public_keys_input
+  let json_string = Core.In_channel.input_all Core.In_channel.stdin in
+  let json =
+    Yojson.Basic.from_string json_string |> change_constraint
+    (* |> remove_public_keys_input
```
Comment (bkase) [inline scripts/archive/ocaml/stitch_introspection.ml]: any reason why commenting out instead of deleting?

**Comment by bkase** [comment]
> should be put it in gitattributes?

this is how you skip diffs on it right? I think we should

File: scripts/archive/ocaml/stitch_introspection.ml (hunk 2 above)
Comment (wu-s-john) [reply scripts/archive/ocaml/stitch_introspection.ml]: The entire file is already deleted

File: src/app/archive/lib/ast.ml [hunk 3]
```diff
@@ -0,0 +1,216 @@
+module Rel_insert_input = struct
+  type ('data, 'on_conflict) t =
+    < data: 'data ; on_conflict: 'on_conflict option >
+
+  type nonrec ('data, 'on_conflict) array = ('data array, 'on_conflict) t
+end
+
+(* Hack: void is an inhabited type. It is used to make empty lists *)
```
[the remark this answers is not in the document]
Comment (Schmavery) [reply src/app/archive/lib/ast.ml]: Uninhabited?

File: src/app/archive/lib/processor.ml [hunk 4]
```diff
@@ -57,10 +59,162 @@ module Make (Config : Graphql_client_lib.Config_intf) = struct
...
+        (compute_with_receipt_chains user_commands_with_time
+           sender_receipt_chains_from_parent_ledger)
+        fee_transfers_with_time
+    in
+    let graphql =
+      Graphql_query.Blocks.Insert.make
+        ~blocks:(Array.of_list [encoded_block])
+        ()
+    in
+    let%bind _ = Client.query_or_error graphql port in
```
Comment (Schmavery) [inline src/app/archive/lib/processor.ml]: Are we throwing away errors?

**Comment by ghost-not-in-the-shell** [comment]
Should you add yourself as the code owner for the newly added files?

**Pull request merged**